File: 1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, n in enumerate(nums):
            tmp = target - n
            if tmp in seen:
                res = [seen[tmp],i]
                seen.clear()
                return res
            else:
                seen[n] = i
        return [-1,-1]

File: 1.cpp
#include <unordered_map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<int> result{};
        std::unordered_map<int, int> seen{};
        for(auto i = 0; i < nums.size(); ++i)
        {   
            int tmp = target - nums[i];
            if(seen.contains(tmp))
            {
                result.push_back(seen[tmp]);
                result.push_back(i);
                return result;
            }
            if(!seen.contains(nums[i]))
            {
                seen[nums[i]] = i;
            }
        }
        return result;
    }
};
